### PR TITLE
fix: adjust the left/right canvas width properly when fullWidthRows is used

### PR DESCRIPTION
**Pull request by xinchao-zhang**
When `fullWidthRows: true` is used in options, and when the viewport width is larger than the canvas width, slick grid does not create rows wide enough to fill the viewport.

This happens because although the total canvas width has been calculated to match that of viewport, the total canvas width is not distributed to the left/right canvas.

To illustrate:

 **Without the fix and with this option: `{ fullWidthRows: true }`** , there is an area of white space at the right side of the canvas (selected row is colored in blue to make it more obvious):
current

**And with this option: `{ fullWidthRows: true,     frozenColumn: 2 }`**:
current_with_col_freeze

Basically visually the fullWidthRows has no effect.

**With the fix and with this option: `{ fullWidthRows: true }`**:
fix
**Same for frozen columns `{ fullWidthRows: true,     frozenColumn: 2 }`**:
fix_with_column_freeze

### CODE REVIEW

File: slick.grid.js [hunk 1]
```diff
@@ -828,7 +828,17 @@ if (typeof Slick === "undefined") {
...
+        var extraWidth = Math.max(totalRowWidth, availableWidth) - totalRowWidth;
+        if (extraWidth > 0) {
+          totalRowWidth += extraWidth;
+          if (hasFrozenColumns()) {
+            canvasWidthR += extraWidth;
+          } else {
+            canvasWidthL += extraWidth;
+          }
+        }
+      }
```
Comment (ghiscoding) [inline slick.grid.js:841]: you're no longer returning any width, so that will break the function

**Comment by ghiscoding** [comment]
@xinchao-zhang 
can you show print screen or animated gif of what that fixes? it's hard to see what would be the difference just in code

**Comment by xinchao-zhang** [comment]
@ghiscoding I've updated my original comment with screenshots

**Comment by ghiscoding** [comment]
ok that's much better to understand, but I wonder why do you think it's better with the fix and what is your use case? 
I kind of prefer without the fix, if I select a row I don't really want it to go outside the columns (same goes for background coloring), isn't it better to just change the viewport width of the grid?

**Comment by xinchao-zhang** [comment]
This renders the data grid in a way more similar to e.g. Excel. In my view it just makes the grid looks more polished. We don't always have the luxury/flexibility to shrink the viewport in order to get rid of the white space. E.g. the grid might be only a part of a larger layout and the size of the grid viewport is decided externally and dynamically (think flexbox). Meanwhile the column set can change dynamically as well (e.g. user can add or remove or resize the columns at runtime) so it is hard to always statically fix the  viewport size. 

But most importantly, what does `fullWidthRows` option supposed to do then lol? Do you mean you would prefer removing the fullWidthRows feature?

**Comment by ghiscoding** [comment]
oh right I forgot you mention that it's with the `fullWidthRows` grid option enabled, that's my bad I read it too quickly (I have never tried that option actually). So in that case I would agree with you then. Let see if @6pac has any opinions on the subject.

**Comment by 6pac** [comment]
Wasn't aware it even existed!

**Comment by 6pac** [comment]
This looks good to me. OK to proceed @ghiscoding ?

**Comment by ghiscoding** [comment]
I'd be ok with it
